src/hotspot/share/opto/gcm.cpp

```diff
@@ -769,6 +773,40 @@ Block* PhaseCFG::insert_anti_dependences(Block* LCA, Node* load, bool verify) {
     }
   }
   worklist_def_use_mem_states.push(nullptr, initial_mem);
+  Block* initial_mem_block = get_block_for_node(initial_mem);
+  assert(initial_mem_block != nullptr, "sanity");
+  if (load->in(0) != nullptr) {
+    // If the load has an explicit control input, walk up the dominator tree
+    // from the early block (inclusive) to the initial memory block
+    // (inclusive). If we in a block find memory Phi(s) that can alias
```

Comment:
"If we in a block find" sounds a little strange.

```suggestion
    // (inclusive). When traversing the blocks, we look for Phi(s) that can alias
```